### ISSUE TITLE
multipart/form-data endpoint doesn't seem to have correct content-type

### ISSUE THREAD

**Issue by fr3fou** [anymaniax/orval#175]
## What are the steps to reproduce this issue?
1. Generating a react-query query on an endpoint that expects `multipart/form-data`

## What happens?
The request that gets sent uses `application/json`

## What were you expecting to happen?
The request to have `multipart/form-data`

## What versions are you using?
**Operating System:** macOS 11.4
**Package Version:** 5.4.9

**Comment by fr3fou** [comment]
I think the generated react-query also seems to be wrong, shouldn't it use `new FormData()` and use that instead?
```ts
export const meAvatarUpload = <TData = WebMediaUploadResponse>(
  brandLogoUploadBody: BrandLogoUploadBody,
  options?: SecondParameter<typeof customInstance>
) => {
  return customInstance<TData>(
    {
      url: `/me/media`,
      method: "post",
      data: brandLogoUploadBody,
    },
    // eslint-disable-next-line
    // @ts-ignore
    options
  )
}

export const useMeAvatarUpload = <
  TData = AsyncReturnType<typeof meAvatarUpload, WebMediaUploadResponse>,
  TError = WebErrorResponse,
  TContext = unknown
>(options?: {
  mutation?: UseMutationOptions<
    TData,
    TError,
    { data: BrandLogoUploadBody },
    TContext
  >
  request?: SecondParameter<typeof customInstance>
}) => {
  const { mutation: mutationOptions, request: requestOptions } = options || {}

  return useMutation<TData, TError, { data: BrandLogoUploadBody }, TContext>(
    (props) => {
      const { data } = props || {}

      return meAvatarUpload<TData>(data, requestOptions)
    },
    mutationOptions
  )
}
```

**Comment by anymaniax** [comment]
I am not sure to understand your example. Can I get the swagger?

**Comment by sladg** [comment]
So I'm dealing with some FormData issue here.
I have this endpoint:
`        "\/token#facebook_access_token": {
            "post": {
                "tags": ["User"],
                "operationId": "postFacebookTokenCredentialsItem",
                "summary": "Get Access Token using Facebook Token (oAuth2)",
                "requestBody": {
                    "description": "Get Access Token",
                    "content": {
                        "multipart\/form-data": {
                            "schema": {
                                "$ref": "#\/components\/schemas\/FacebookCredentials"
                            }
                        }
                    }
                },
                "responses": {
                    "200": {
                        "description": "Get access token",
                        "content": {
                            "application\/json": {
                                "schema": {
                                    "$ref": "#\/components\/schemas\/Token"
                                }
                            }
                        }
                    },
                    "400": {
                        "description": "Invalid request"
                    },
                    "404": {
                        "description": "Not found"
                    },
                    "409": {
                        "description": "Email is already used"
                    }
                }
            }
        },
`

However, the output looks like this:
`
export interface FacebookCredentials {
  grant_type?: string;
  client_id?: string;
  client_secret?: string;
  facebook_access_token?: string;
  facebook_user_id?: string;
  facebook_token_expires_in?: string;
  create_account?: boolean;
  types?: string;
  type_tag?: FacebookCredentialsTypeTag;
}

export const postFacebookTokenCredentialsItem = <TData = Token>(
  facebookCredentials: FacebookCredentials,
  options?: SecondParameter<typeof axiosClient>) => {
  return axiosClient<TData>(
    {
      url: `/token#facebook_access_token`, method: 'post',
      data: facebookCredentials
    },
    // eslint-disable-next-line
    // @ts-ignore
    options);
}
`

So it does not respect FormData and just converts it into a normal JSON body. Which is far from great.

**Comment by sladg** [comment]
So I'm dealing with some FormData issue here.
I have this endpoint:
```      "\/token#facebook_access_token": {
            "post": {
                "tags": ["User"],
                "operationId": "postFacebookTokenCredentialsItem",
                "summary": "Get Access Token using Facebook Token (oAuth2)",
                "requestBody": {
                    "description": "Get Access Token",
                    "content": {
                        "multipart\/form-data": {
                            "schema": {
                                "$ref": "#\/components\/schemas\/FacebookCredentials"
                            }
                        }
                    }
                },
                "responses": {
                    "200": {
                        "description": "Get access token",
                        "content": {
                            "application\/json": {
                                "schema": {
                                    "$ref": "#\/components\/schemas\/Token"
                                }
                            }
                        }
                    },
                    "400": {
                        "description": "Invalid request"
                    },
                    "404": {
                        "description": "Not found"
                    },
                    "409": {
                        "description": "Email is already used"
                    }
                }
            }
        },
```

However, the output looks like this:

```
export interface FacebookCredentials {
  grant_type?: string;
  client_id?: string;
  client_secret?: string;
  facebook_access_token?: string;
  facebook_user_id?: string;
  facebook_token_expires_in?: string;
  create_account?: boolean;
  types?: string;
  type_tag?: FacebookCredentialsTypeTag;
}

export const postFacebookTokenCredentialsItem = <TData = Token>(
  facebookCredentials: FacebookCredentials,
  options?: SecondParameter<typeof axiosClient>) => {
  return axiosClient<TData>(
    {
      url: `/token#facebook_access_token`, method: 'post',
      data: facebookCredentials
    },
    // eslint-disable-next-line
    // @ts-ignore
    options);
}
```

So it does not respect FormData and just converts it into a normal JSON body. Which is far from great.

**Comment by anymaniax** [comment]
Since you are using a mutator you need to set it yourself. But that's true that I could automatically add it like without mutator.

**Comment by sladg** [comment]
@anymaniax it would be amazing to have support for Form Data out-of-box. Mutator can fix it, but I would need to manually map all those data, which is not super sustainable.

**Comment by anymaniax** [comment]
@sladg Orval already has that when your aren't using Mutator. But maybe I should add a toggle or something to activate it with mutator too. It's also really opinionated the way you create the formData that why didn't add it directly with Mutator. I will think about it

**Comment by anymaniax** [comment]
You be cool to have the same think as the Mutator but just for formData no?

**Comment by sladg** [comment]
> @sladg Orval already has that when your aren't using Mutator. But maybe I should add a toggle or something to activate it with mutator too. It's also really opinionated the way you create the formData that why I didn't add it directly with Mutator. I will think about it

I see. I'm thinking that the best way might be a parameter in the config?

In my case, the mutator manages retries and refreshing token on 401, so having separate config just for formData seems like overhead. I cannot think of a situation when I would want to handle formData myself and not use out-of-box solution.

**Comment by anymaniax** [comment]
should be fixed in `5.4.13` and you have a new property formData: boolean 🎉. I will add the feature to override the generation of form data in the next big version

**Comment by sladg** [comment]
@anymaniax I confirm that now FormData works as expected. Thanks a lot! We can probably close the issue.

**Comment by fr3fou** [comment]
> should be fixed in `5.4.13` and you have a new property formData: boolean 🎉. I will add the feature to override the generation of form data in the next big version

How am I supposed to use that property? I only need it for a few endpoints

**Comment by anymaniax** [comment]
@fr3fou normally it's automatic if you put that your endpoint is a `multipart/form-data` and the boolean is only useful if you want to disable the automatic generation of formData
